### PR TITLE
fix(UI): Button alignment in email signup

### LINKED ISSUE #52131: Alignment Issue: Button in Email Sign-up Section
### Describe the Issue

In the email signup page the button appears to be misaligned resulting in an unbalanced layout.
es

### Affected Page

https://www.freecodecamp.org/email-sign-up/

### Steps to Reproduce

1. Sign in as a new user on https://www.freecodecamp.org ( If you have accepted the Privacy Terms ).
2. Upon successful sign-in, you will be redirected to the email signup screen.
3. Scroll down until you reach the email subscription section.
4. Observe that the button is misaligned.

### Expected behavior

The button should be aligned properly within its container.
image

### Screenshots

_No response_

### System

- Device: [Laptop]
- Browser: [Edge, Chrome]

### Additional context

I would like to work on this

### PR DESCRIPTION
**Pull request by JeevaRamanathan**
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52131 

The change involves fixing the misalignment of the button in the email sign up page.
Screenshots: 
image

### CODE REVIEW

File: client/src/pages/email-sign-up.tsx
Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.tsx:64]: Hi @huyenltnguyen ,
 I understand your concern about avoiding the usage of CSS overrides in the code. I have another approach. Instead of modifying the CSS, we can add the "flex" class to the row component like this: ```<Row className="flex">```.

This will result in a slight difference in the layout in mobile view. The two buttons will appear on the same line instead of being stacked vertically.

Do you think this alternative solution could be helpful?

Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.tsx:64]: + I have identified one more issue along with this. Due to the ```Row``` component which has margin of ```-15px``` by default, it is causing an empty space and as a result footer is not properly aligned in mobile view.
image


So alternate suggestion is to add  ```<Container className='flex'>```  instead  ```Row```, which will solve both button alignment and extra space in footer without CSS. 

image

Would this approach be acceptable in your view?

File: client/src/pages/email-sign-up.css
Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.css:6]: Sure, I will add it.  Additionally as mentioned in comment https://github.com/freeCodeCamp/freeCodeCamp/pull/52132#discussion_r1375003096 , the ```Row``` component with a default margin of -15px adds extra space in the footer, Can I modify that into ```<Contanier classname="email-list-opt">``` as well to achieve the proper alignment?

Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.css:6]: Hi @huyenltnguyen, thanks for your response.
1. Upon going through most of the pages, as I have observed, I can see this only on this page, at least so far.
<hr/>

2.
###
    import { Col, Row } from '@freecodecamp/react-bootstrap;
    ...
    <Row>
      <Col>Button</Col>
      <Col>Button</Col>
     </Row>

 So with the above line the buttons are stacked vertically in mobile as in here https://user-images.githubusercontent.com/64531160/278748565-c07d93e1-e4b6-409c-98e5-7f33bdc43bac.png


And the new change
    ```import { Container, Col, Row } from '@freecodecamp/ui';```
   ```<Container className='email-list-opt'>```

 as well results in the same of button stacked vertically except the additional space.
image


So it is fine to go with 2nd approach?

Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.css:6]: Oh ok, I was testing it with Chrome and Edge and seeing the gap on refresh too. But even in the current original version (ie. in freecodecamp.com/email-sign-up ) as well as seeing the gap and also there is no space in the left side of the button in chrome [tested on mobile] even on

Comment (JeevaRamanathan) [reply client/src/pages/email-sign-up.css:6]: Oh ok, I was testing it with Chrome and Edge. 
But even in the current original version (ie https://freecodecamp.com/email-sign-up with code ```<Row>``` ) as well I'm seeing the gap and also there is no space in the left side of the button [tested on mobile directly in chrome browser] even on refresh. So hoping that obv in mobile that issue may still persist if it's row due to margin -15px.
Image: https://tinyurl.com/3n7wnwh8


But on settings screen https://freecodecamp.org/settings I could see a equal amount of gap on both sides on save button.

And yes ```container```  makes the button width smaller as it has 15px margin by default.

So ```<div className='email-sign-opt'>``` give a similar view as in original version/settings screen.

So please once again have a look at this 
testing on your screens and the final decision is yours 😅

**Review by huyenltnguyen — APPROVED** [review]
@JeevaRamanathan thank you for the PR and for bearing with me!

I've tested locally using the combination of `Row` and `Col` from both `ui-components` and `@freecodecamp/react-bootstrap`, and I didn't see a gap between the footer and the edge of the screen.

Here is the result (tested on both Chrome and Firefox):

| Desktop | Mobile |
| --- | --- |
| Screenshot 2023-11-06 at 16 10 40 | Screenshot 2023-11-06 at 16 07 39 |

With that, I updated the PR to switch back to the `Row` component. Let's wait for another maintainer to help us test and give the final verdict 😄

**Comment by JeevaRamanathan** [comment]
@huyenltnguyen  Sorry for pulling your leg or dragging this conversation further😅, but I still have a doubt that I'd like to clarify.
So in the screenshot that you have shared above, if you see that profile icon it's partially hidden which is also causing that space in the footer.
image

And if you see the below screenshot (settings screen), its having equal margins on save button and also profile icon is clearly visible.
image.

So it that expected to have no margins in Yes/No button in email-sign-up screen or could you please clarify?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
